lib/ssh/src/ssh.hrl

```diff
@@ -152,70 +152,91 @@ to run any subsystems.
-                            'diffie-hellman-group18-sha512' |
-                            'diffie-hellman-group14-sha256' |
-                            'diffie-hellman-group14-sha1' |
-                            'diffie-hellman-group-exchange-sha1' |
-                            'diffie-hellman-group1-sha1'
-                            .
+-type kex_alg()  ::
+        'diffie-hellman-group-exchange-sha256' |
+        'diffie-hellman-group14-sha256' |
+        'diffie-hellman-group16-sha512' |
+        'diffie-hellman-group18-sha512' |
+        'curve25519-sha256' |
+        'curve25519-sha256@libssh.org' |
+        'curve448-sha512' |
+        'ecdh-sha2-nistp256' |
+        'ecdh-sha2-nistp384' |
+        'ecdh-sha2-nistp521' |
+        legacy_kex_alg().
+
+-doc(#{title => <<"Legacy Algorithms">>}).
+-type legacy_kex_alg()  ::
+        %%  Gone in OpenSSH 7.3.p1
+        'diffie-hellman-group1-sha1' |
+        %%  Gone in OpenSSH 8.2
+        'diffie-hellman-group14-sha1' |
+        'diffie-hellman-group-exchange-sha1'.
 
 -doc(#{title => <<"Common Options">>}).
--type pubkey_alg()       :: 'ssh-ed25519' |
-                            'ssh-ed448' |
-                            'ecdsa-sha2-nistp521' |
-                            'ecdsa-sha2-nistp384' |
-                            'ecdsa-sha2-nistp256' |
-                            'rsa-sha2-512' |
-                            'rsa-sha2-256' |
-                            'ssh-rsa' |
-                            'ssh-dss'
-                            .
+-type pubkey_alg()  ::
+        'ecdsa-sha2-nistp256' |
+        'ecdsa-sha2-nistp384' |
+        'ecdsa-sha2-nistp521' |
+        'ssh-ed25519' |
+        'ssh-ed448' |
+        'rsa-sha2-256' |
+        'rsa-sha2-512' |
+        legacy_pubkey_alg().
+
+-doc(#{title => <<"Legacy Algorithms">>}).
+-type legacy_pubkey_alg()  ::
+        'ssh-rsa' |
+        %% Gone in OpenSSH 7.3.p1:
+        'ssh-dss'.
 
 -doc(#{title => <<"Common Options">>}).
--type cipher_alg()       :: 'aes256-gcm@openssh.com' |
-                            'aes256-ctr' |
-                            'aes192-ctr' |
-                            'aes128-gcm@openssh.com' |
-                            'aes128-ctr' |
-                            'AEAD_AES_256_GCM' |
-                            'AEAD_AES_128_GCM' |
-                            'chacha20-poly1305@openssh.com' |
-                            'aes256-cbc' |
-                            'aes192-cbc' |
-                            'aes128-cbc' |
-                            '3des-cbc'
-                            .
+-type cipher_alg()  ::
+        'aes128-ctr' |
+        'aes128-gcm@openssh.com' |
+        'aes192-ctr' |
+        'aes256-ctr' |
+        'aes256-gcm@openssh.com' |
+        'chacha20-poly1305@openssh.com' |
+        legacy_cipher_alg()
+        .
+
+-doc(#{title => <<"Legacy Algorithms">>}).
+-type legacy_cipher_alg()  ::
+        'AEAD_AES_128_GCM' |
```

Comment:
Maybe the AEAD algorithms should have a own category as I think it is something that in the long run should be put back, once the terrapin fallback behavior is not default.

Reply:
good suggestion. I've applied it in amended commit.
reason for having them disabled is explained in https://github.com/erlang/otp/commit/a895fc7303497f1795cf49360980abeb68be2223

I've included reference in code comment and a brief reason wrap-up.

Reply:
Ok so the reason was an other than I thought I remembered. But like the new doc.